### PR TITLE
add qwen3-moe optimization

### PR DESCRIPTION
**Pull request by shiyuan680**
### What this PR does / why we need it?
origin qwen3_moe loss alltoall operation which result fault resultt, in this pr reuse some optimizations from deepseek.

### How was this patch tested?

test in 235b
parallelism           tps        open
dp16tp2ep32       160        close
dp16tp2ep32       192       on
dp8tp4ep32       76        close
dp8tp4ep32       128       on

### CODE REVIEW

File: tests/e2e/singlecard/test_offline_inference.py [hunk 1]
```diff
@@ -35,6 +35,7 @@
 MODELS = [
     "Qwen/Qwen2.5-0.5B-Instruct",
     "Qwen/Qwen3-0.6B-Base",
+    "Qwen/Qwen3-30B-A3B",
```
Comment (Yikun) [inline tests/e2e/singlecard/test_offline_inference.py]: This file is too huge and will cost lots of time to run ci, please try to used reduce layer model: https://vllm-ascend.readthedocs.io/en/latest/developer_guide/contribution/testing.html#e2e-test-example

File: vllm_ascend/models/qwen3_moe.py [hunk 2]
```diff
@@ -33,3 +57,89 @@ class CustomQwen3MoeForCausalLM(Qwen3MoeForCausalLM):
         "experts":
         ["experts.0.gate_proj", "experts.0.up_proj", "experts.0.down_proj"],
     }
+
+
+class AscendQwen3MoeSparseMoeBlock(nn.Module):
```
Comment (Yikun) [inline vllm_ascend/models/qwen3_moe.py:54]: Please add ut for this: https://vllm-ascend.readthedocs.io/en/latest/developer_guide/contribution/testing.html

**Review by Yikun — COMMENTED** [review]
Please update commits msg more meanful, such as mention what kind of change apply compare to upstream impementation and performance test results

**Comment by leograssroot** [comment]
dp切分或者dp+tp切分场景下，执行客户的其他moe模型也会有精度问题，会有一句话不断重复回答的问题

**Comment by shiyuan680** [comment]
> dp切分或者dp+tp切分场景下，执行客户的其他moe模型也会有精度问题，会有一句话不断重复回答的问题
是qwen3-moe模型吗

**Comment by shiyuan680** [comment]
> dp切分或者dp+tp切分场景下，执行客户的其他moe模型也会有精度问题，会有一句话不断重复回答的问题

是qwen3-moe模型吗，这个只是针对qwen3-moe模型的修复

**Comment by leograssroot** [comment]
> > dp切分或者dp+tp切分场景下，执行客户的其他moe模型也会有精度问题，会有一句话不断重复回答的问题
> 
> 是qwen3-moe模型吗，这个只是针对qwen3-moe模型的修复

不是qwen3，这个切分不是通用的问题么

**Comment by leograssroot** [comment]
> > dp切分或者dp+tp切分场景下，执行客户的其他moe模型也会有精度问题，会有一句话不断重复回答的问题
> 
> 是qwen3-moe模型吗，这个只是针对qwen3-moe模型的修复

https://github.com/vllm-project/vllm-ascend/issues/1597
